doc/joss/paper.md

```diff
@@ -0,0 +1,141 @@
+without introducing performance degradations. 
+
+Ginkgo is also a part of the xSDK effort [@xsdk] which aims to provide infrastructure
+for and interoperability of a collection of related and complementary software 
+elements to foster rapid and efficient development of scientific applications
+using High Performance Computing. Within this effort, we support interoperability with
+application libraries such as `deal.ii`[@dealii] and `mfem`[@mfem]. Ginkgo provides
+wrappers within these two libraries so that they can take advantage of the features of
+Ginkgo. 
+
+# Features
+
+As sparse linear algebra is one of the main focus of Ginkgo, we provide a variety of 
+sparse matrix formats such as COO, CSR, ELL, HYBRID and SELLP along with highly tuned
+Sparse Matrix Vector product (SpMV) kernels. The SpMV kernel is a key building 
+blocks of vitually all iterative solvers and typically accounts for a significant 
+fraction of the application runtime. Additionally, we also provide vhigh performance 
+conversion routines between the different formats enhancing their flexibility.
+
+Ginkgo provides multiple iterative solvers such as the Krylov subspace
+methods: Conjugate gradient (CG), Flexible Conjugate Gradient (FCG), Bi-Conjugate
+Gradient (BiCG) and its stabilized version (Bi-CGSTAB), Generalized Minimal
+residual method (GMRES) and more generic methods such as Iterative Refinement,
+which forms the basis of many relaxation methods. Ginkgo also features support for 
+direct and iterative triangular solves within incomplete factorization preconditioners.
+
+Ginkgo features some of the best preconditioners such as the Block Jacobi
+preconditioner with support for a version which reduces pressure on the memory bandwidth 
+by dynamically adapting the memory precision to the numerical requirements.
+This [@adaptive-bj] has been shown to be very efficient for problems with a block 
+structure.
+Ginkgo also features highly-parallel incomplete factorization preconditioners such as 
+the ParILU and the ParILUT preconditioners [@parilut].
+
+
+# Software extensibility and sustainability.
+
+Ginkgo is extensible in terms of linear algebra solvers, preconditioners and
+matrix formats. Basing on modern C++ (C++11 standard), various language features
+such as data abstraction, generic programming and automatic memory management are
+leveraged to enhance the performance of the library while still maintaining ease of 
+use and maintenance. 
+
+The Ginkgo library is constructed around two principal design concepts. The first
+one is the class and object-oriented design which aims to provide an easy to use 
+interface, common for all the devices; and the second part consists of the low level
+device specific kernels. These low level kernels are optimized for the specific device
+and make use of C++ features such as templates to generate high-performance kernels 
+for a wide variety of parameters. 
+
```

Comment:
Should we mention our loggers here? I would consider them a really nice and important feature if you want to gain a bigger insight into the execution of our algorithms

Reply:
We do mention in the performance and benchmarking section. But we can move it here.

Reply:
Ah right, skipped over that sentence. Then nevermind :)